src/Classes/ServiceAPI/Profile.php

```diff
@@ -72,14 +77,22 @@ public static function getMembersForYear($year)
     {
         self::wakeup();
 
-        return self::$db->fetchAll(
-            'SELECT member.memberid, sname || \', \' || fname AS name, l_college.descr AS college, paid, email, eduroam
+        /* Adding nickname can cause issues if the nickname is null;
+        * If you work out a way to have a default value for null values in SQL,
+        * message me on slack @Ren Herring
```

Comment:
this really should be done outside of SQL rather than trying to bodge it here (and all the other cases with similar logic (it should be broken out into a method on the user object, rather than repeating the same sql concatenation))

Reply:
So I imagine this was originally done in SQL because `getMembersForYear` *could* fetch a load of members who aren't in the cache, so a `MyRadio_User::resultSetToObjArray()` could be slow as it'd have to fetch each of them one by one. But I agree with Ash, may be worth switching to do it here.

(but if you really want to do this in SQL,
```sql
SELECT
CASE WHEN nname IS NULL
THEN fname || ' ' || sname
ELSE fname || ' "' || nname || '" ' || sname
END AS name
```
)

Reply:
how bad would `MyRadio_User::resultSetToObjectArray()` be for preformance or would there be a better alternative